packages/client/src/DataCommonsClient.ts

```diff
@@ -0,0 +1,480 @@
+    return csvLines.join("\n");
+  }
+
+  async getGeoJSON(params: GetGeoJSONParams): Promise<FeatureCollection> {
+    const geoJsonProperty = params.geoJsonProperty || "geoJsonCoordinatesDP1";
+    const dataRows = await this.getDataRows({
+      ...params,
+      entityProps: [
+        geoJsonProperty,
+        ...(params.entityProps ? params.entityProps : DEFAULT_ENTITY_PROPS),
+      ],
+    });
+
+    const geoJson: FeatureCollection = {
+      type: "FeatureCollection",
+      features: dataRows
+        .filter((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`];
+          if (geometryString !== "string") {
+            return true;
+          }
+        })
+        .map((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`] as string;
+          const geometry = JSON.parse(geometryString) as Geometry;
+          const properties = { ...dataRow };
+          delete properties[`entity.${geoJsonProperty}`];
+          const feature: Feature = {
+            type: "Feature",
+            properties,
+            geometry,
+          };
+          return feature;
+        }),
+    };
+    return geoJson;
+  }
+
+  /**
+   * Fetches data commons variable values about an entity or entities.
+   * @param params
+   * @returns
+   */
+  async getDataRows(params: GetDataRowsParams): Promise<DataRow[]> {
+    // Fetch variable observations
+    const pointApiResponse =
+      "parentEntity" in params
+        ? await this.getObservationsPointWithin(params)
+        : await this.getObservationsPoint(params);
+    if (!pointApiResponse) {
+      return [];
+    }
+    const entityDcids =
+      this.getEntityDcidsFromPointApiResponse(pointApiResponse);
+    // Fetch relevant entity and variable property values
+    const entityPropValues = await this.getNodePropValues(
+      entityDcids,
+      params.entityProps || DEFAULT_ENTITY_PROPS
+    );
+    const variablePropValues = await this.getNodePropValues(
+      params.variables,
+      params.variableProps || DEFAULT_VARIABLE_PROPS
+    );
+
+    // Fetch population data for per capita calculations
+    let populationObservations: SeriesApiResponse = { data: {}, facets: {} };
+    if (!_.isEmpty(params.perCapitaVariables)) {
+      populationObservations =
+        "parentEntity" in params
+          ? await this.getObservationsSeriesWithin({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            })
+          : await this.getObservationsSeries({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            });
+    }
+
+    const dataRows = this.getDataRowsFromObservations(
+      entityDcids,
+      params.variables,
+      pointApiResponse,
+      entityPropValues,
+      variablePropValues,
+      populationObservations
+    );
+
+    return Promise.resolve(dataRows);
+  }
+
+  async getFirstNodeValues(params: {
```

Comment:
add a function comment about what this does? Just reading the name of the function is not super clear

Reply:
Added, and also commented `getNodePropvals` which might help the clarify things as well